hyperspy/tests/io/test_fei.py

```diff
@@ -0,0 +1,287 @@
+# (at your option) any later version.
+#
+#  HyperSpy is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with  HyperSpy.  If not, see <http://www.gnu.org/licenses/>.
+
+import os
+import nose.tools as nt
+
+from hyperspy.io import load
+from hyperspy.io_plugins.fei import load_ser_file
+
+my_path = os.path.dirname(__file__)
+
+class test_fei_reader():
+    def setUp(self):
+        self.dirpathold = os.path.join(my_path, 'FEI_old')
+        self.dirpathnew = os.path.join(my_path, 'FEI_new')
+        
+    def test_load_emi_old_new_format(self, verbose=True):
+        # TIA old format
+        fname0 = os.path.join(self.dirpathold, '64x64_TEM_images_acquire.emi')
+        load(fname0, verbose=verbose)
+        # TIA new format
+        fname1 = os.path.join(self.dirpathnew, '128x128_TEM_acquire-sum1.emi')
+        load(fname1, verbose=verbose)
+        # TIA new format 1Go file
+#        fname2= os.path.join('/mnt/data/test/64bits', 'CCD Preview-250-beam_damaged.emi')
+#        load(fname2, verbose=verbose)
+        # TIA new format 4Go file
+#        fname2= os.path.join('/mnt/data/test/64bits', '11.22.31 CCD Preview-250-bean_damaged-bin2-0.2s.emi')
+#        load(fname2, verbose=verbose)
+        
+    def test_load_ser_reader_old_new_format(self, verbose=True):
+        # test TIA old format
+        fname0 = os.path.join(self.dirpathold, '64x64_TEM_images_acquire_1.ser')
+        header0, data0 = load_ser_file(fname0, verbose=verbose)
+        nt.assert_equal(header0['SeriesVersion'], 528)    
+        # test TIA new format
+        fname1 = os.path.join(self.dirpathnew, '128x128_TEM_acquire-sum1_1.ser')
+        header1, data1 = load_ser_file(fname1, verbose=verbose)
+        nt.assert_equal(header1['SeriesVersion'], 544)
+
+    def test_load_diffraction_point(self, verbose=True):
+        fname0 = os.path.join(self.dirpathold, '64x64_diffraction_acquire.emi')
+        s0 = load(fname0, verbose=verbose)
+        nt.assert_equal(s0.data.shape, (64, 64))
+        nt.assert_equal(s0.metadata.Signal.record_by, 'image')
+        nt.assert_equal(s0.metadata.Acquisition_instrument.TEM.acquisition_mode, 'TEM Diffraction')
+        nt.assert_almost_equal(s0.axes_manager[0].scale, 0.10157, places=5)
+        nt.assert_equal(s0.axes_manager[0].units, '1/nm')
+        nt.assert_almost_equal(s0.axes_manager[1].scale, 0.10157, places=5)
+        nt.assert_equal(s0.axes_manager[1].units, '1/nm')
+        
+    def test_load_diffraction_line_scan(self, verbose=True):
+        fname0 = os.path.join(self.dirpathnew, '16x16-line_profile_horizontal_5x128x128_EDS.emi')
+        s0 = load(fname0, verbose=verbose)
+        #s0[1] contains EDS
```

Comment:
On my system (fedora 23), the first item in the list is the diffraction patterns and the 2nd one is the EDS... which doesn't make sense, I don't understand... however, travis gets it right!